pubtools/pulplib/_impl/client/client.py

```diff
@@ -363,3 +372,8 @@ def _delete_upload_request(self, upload_id):
 
         LOG.debug("Deleting upload request %s", upload_id)
         return self._request_executor.submit(self._do_request, method="DELETE", url=url)
+
+    def _do_get_maintenance(self):
+        url = os.path.join(self._url, "pulp/isos/redhat-maintenance/repos.json")
+
+        return self._request_executor.submit(self._do_request, method="GET", url=url)
```

Comment:
I've just had the realization that this is going to use the same requests session as used for the Pulp API, which means the same "auth".  This is a bit weird because the auth is only required for requests to the API, and requests for content like this should not need any authentication.

I *guess* that this shouldn't cause a problem and it's OK to pass the auth even when not required, but could you please check if you haven't already tested this against the real Pulp servers with auth?

Reply:
Yeah, I checked on the rhsm-pulp-dev, it works fine, at least for getting the file.